internal/controller/backup_controller.go

```diff
@@ -287,7 +287,10 @@ func (r *BackupReconciler) isValidBackupRunning(
 		return false, fmt.Errorf("unknown.spec.target received: %s", backup.Spec.Target)
 	}
 
-	containerIsNotRestarted := backup.Status.InstanceID.ContainerID == pod.Status.ContainerStatuses[0].ContainerID
+	var containerIsNotRestarted bool
+	if utils.PodHasContainerStatuses(pod) {
+		containerIsNotRestarted = backup.Status.InstanceID.ContainerID == pod.Status.ContainerStatuses[0].ContainerID
+	}
```

Comment:
```suggestion
	containerIsNotRestarted := utils.PodHasContainerStatuses(pod) && backup.Status.InstanceID.ContainerID == pod.Status.ContainerStatuses[0].ContainerID
```

What do you think of this @gabriele-wolfox ? I find it more readable